fbw-a380x/src/base/flybywire-aircraft-a380-842/SimObjects/AirPlanes/FlyByWire_A380_842/model/behaviour/glareshield.xml

```diff
@@ -8,24 +8,35 @@
         <!-- ATC MSG -->
         <UseTemplate Name="FBW_Push_Toggle">
             <NODE_ID>PUSH_GLARESHIELD_CS_ATCMSG</NODE_ID>
-            <TOOLTIPID>ACKNOWLEDGE ATC MSG (INOP.)</TOOLTIPID>
+            <TOOLTIPID>ACKNOWLEDGE ATC MSG</TOOLTIPID>
+            <LEFT_SINGLE_CODE>1 (&gt;L:A32NX_DCDU_ATC_MSG_ACK)</LEFT_SINGLE_CODE>
+            <SEQ1_CODE>(L:A32NX_DCDU_ATC_MSG_WAITING)</SEQ1_CODE>
+            <DISABLE_SEQ2/>
         </UseTemplate>
 
         <!-- AUTO LAND -->
         <UseTemplate Name="FBW_Push_Toggle">
             <NODE_ID>PUSH_GLARESHIELD_CS_AUTOLAND</NODE_ID>
+            <INDICATOR_CODE>(L:A32NX_AUTOPILOT_AUTOLAND_WARNING, bool) (E:ABSOLUTE TIME, seconds) 1 % 0.4 &gt; and</INDICATOR_CODE>
+            <DISABLE_SEQ2/>
         </UseTemplate>
 
         <!-- MASTER WARN -->
-        <UseTemplate Name="FBW_Push_Toggle">
+        <UseTemplate Name="FBW_Push_Held">
             <NODE_ID>PUSH_GLARESHIELD_CS_MASTERWARN</NODE_ID>
-            <TOOLTIPID>ACKNOWLEDGE MASTER WARNING (INOP.)</TOOLTIPID>
+            <TOOLTIPID>ACKNOWLEDGE MASTER WARNING</TOOLTIPID>
+            <HOLD_SIMVAR>L:PUSH_AUTOPILOT_MASTERAWARN_L</HOLD_SIMVAR>
+            <SEQ1_CODE>(L:A32NX_MASTER_WARNING)</SEQ1_CODE>
+            <DISABLE_SEQ2/>
```

Comment:
Would we also want to add SEQ1_POWERED? 
`<SEQ1_POWERED>(L:A32NX_ELEC_DC_ESS_BUS_IS_POWERED)</SEQ1_POWERED>`

Reply:
We don't have this for the other buttons that are already hooked up and I don't have this kind of info for the A380.